src/App.jsx

```diff
@@ -14,33 +17,88 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABETICALLY = 'Alphabetically';
+const SORT_FIELD_LENGTH = 'Length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getSortedGoods(goods, sortField, reverse) {
+  let sortedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  switch (sortField) {
+    case SORT_FIELD_ALPHABETICALLY:
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+      break;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    case SORT_FIELD_LENGTH:
+      sortedGoods.sort((a, b) => a.length - b.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (reverse) {
+    sortedGoods = sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const visibleGoods = getSortedGoods(goodsFromServer, sortField, reverse);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn({
+            'button is-info ': sortField === SORT_FIELD_ALPHABETICALLY,
+            'button is-info is-light': sortField !== SORT_FIELD_ALPHABETICALLY,
+          })}
```

Comment:
You ar ebit misusing `classnames`

```suggestion
          className={cn('button', 'is-info', { is-light': sortField !== SORT_FIELD_ALPHABETICALLY} )}
```

Reply:
Please, mind the other places too